pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -663,6 +666,7 @@ type Template struct {
 	HTTP *HTTP `json:"http,omitempty" protobuf:"bytes,42,opt,name=http"`
 
 	// Plugin is a plugin template
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
Why do we need this?

Reply:
Because plugins are defined as free-form objects. See https://github.com/argoproj/argo-workflows/issues/7393

Without this, the `TestExecutorPluginsSuite/TestTemplateExecutor` test fails:
```
Submitting workflow  template-executor-
W0111 01:58:53.394874   40529 warnings.go:70] unknown field "spec.templates[0].plugin.hello"
    executor_plugins_test.go:29: expected exactly one key, got 0
=== FAIL: ExecutorPluginsSuite/TestTemplateExecutor
```

Reply:
I left some comments in https://github.com/argoproj/argo-workflows/commit/987bb2e71fccdbe9b58de91ac02ba64911b6c466 explaining this